torch/csrc/dynamo/cpp_shim.cpp

```diff
@@ -14,6 +13,24 @@ _PytorchRecordFunctionState* _pytorch_record_function_enter(const char* name) {
   return state;
 }
 
+static inline _PytorchRecordFunctionState*
+_pytorch_record_function_enter_with_kwinputs(
+    const char* name,
+    const std::unordered_map<std::string, c10::IValue>* kwargs) {
+  _PytorchRecordFunctionState* state = new _PytorchRecordFunctionState();
+  std::vector<c10::IValue> args;
+  state->guard.before(name, &args, kwargs);
+  return state;
+}
+
+_PytorchRecordFunctionState* _pytorch_record_function_enter_with_context(
+    const char* name,
+    const char* context) {
+  auto map = std::make_unique<std::unordered_map<std::string, c10::IValue>>();
```

Comment:
Does this need to be a unique_ptr? Can we not just make it a stack-allocated unordered_map?

Reply:
updated